src/library/scala/collection/StringOps.scala

```diff
@@ -1142,11 +1140,25 @@ final class StringOps(private val s: String) extends AnyVal {
     *
     *  @param  xs     the array to fill.
     *  @param  start  the starting index.
-    *  @param  len    the maximal number of elements to copy.
     */
-  @`inline` def copyToArray(xs: Array[Char], start: Int): Unit =
+  @`inline` def copyToArray(xs: Array[Char], start: Int): Int =
     copyToArray(xs, start, Int.MaxValue)
 
+  /** Copy chars of this string to an array.
+    * Fills the given array `xs` starting at index `start` with at most `len` chars.
+    * Copying will stop once either the entire string has been copied,
+    * or the end of the array is reached or `len` chars have been copied.
+    *
+    *  @param  xs     the array to fill.
+    *  @param  start  the starting index.
+    *  @param  len    the maximal number of elements to copy.
+    */
+  def copyToArray(xs: Array[Char], start: Int, len: Int): Int = {
+    val written = max(min(min(s.length, len), xs.length - start), 0)
+    s.getChars(0, written, xs, start)
```

Comment:
We could event put this line within an `if (written > 0)`

Reply:
👍